src/Microsoft.Dnx.Watcher.Core/ProjectWatcher.cs

```diff
@@ -0,0 +1,286 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Dnx.Compilation;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Dnx.Runtime.Common.Impl;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.Dnx.Watcher.Core
+{
+    public class ProjectWatcher
+    {
+        private readonly Func<string, IFileWatcher> _fileWatcherFactory;
+        private readonly IProjectGraphProvider _projectGraphProvider;
```

Comment:
Read the lock file to figure out the projects. Since you don't have the target framework just watch all projects listed in the lock file. Copy over the logic that figures out the file patterns for a given project file and you're done!